### PR TITLE
Fix the PluginInfoCollector to check RubyGems for the gem_name

### PR DESCRIPTION
**Pull request by mfurtak**
Previously this was checking RubyGems to see if the `plugin_name` was already taken, but this does not contain the 'fastlane-plugin-' prefix, and is not the full final name of the gem.

Fixes up the tests to match this behavior, and provides a helper method for setting up the stubbed web request.

### CODE REVIEW

File: fastlane/lib/fastlane/plugins/plugin_info_collector.rb [hunk 1]
```diff
@@ -37,9 +37,11 @@ def collect_plugin_name(initial_name = nil)
 
         break if plugin_name_valid?(plugin_name)
 
-        if plugin_name_taken?(plugin_name)
-          # Plugin name is already taken on RubyGems
-          @ui.message("\nPlugin name '#{plugin_name}' is already taken on RubyGems, please choose a different one.")
+        gem_name = PluginManager::FASTLANE_PLUGIN_PREFIX + plugin_name
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/plugin_info_collector.rb]: At this point, might `plugin_name` contain the prefix already? What if the user enters `fastlane-plugin-yolo`?

File: fastlane/spec/plugins_specs/plugin_info_collector_spec.rb [hunk 2]
```diff
@@ -9,9 +9,7 @@
 
   before do
     ["my plugin", "test_name", "my_", "fastlane-whatever", "whatever"].each do |current|
-      stub_request(:get, "https://rubygems.org/api/v1/gems/#{current}.json").
-        with(headers: {'Accept' => '*/*', 'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent' => 'Ruby'}).
-        to_return(status: 200, body: nil, headers: {})
+      stub_plugin_exists_on_rubygems(current, false)
```
Comment (KrauseFx) [inline fastlane/spec/plugins_specs/plugin_info_collector_spec.rb]: Nice, I love this! 👍

File: fastlane/lib/fastlane/plugins/plugin_info_collector.rb (hunk 1 above)
Comment (mfurtak) [inline fastlane/lib/fastlane/plugins/plugin_info_collector.rb]: Good point, will fix

**Comment by KrauseFx** [comment]
This looks great, however I'm not yet sure what happens when the user enters `fastlane-plugin-yolo`. Does the complete workflow work as expected (e.g. the action being called `yolo`)

Ready to merge from my side 👍

**Comment by mfurtak** [comment]
Yes, the info collector already handles name input like 'fastlane-plugin-yolo' and it detects it as invalid, but suggests 'yolo' to the user as the valid name.

Then, this change will ensure that RubyGems does not already contain the gem 'fastlane-plugin-yolo' (It was previously checking for 'yolo')

**Comment by KrauseFx** [comment]
Perfect 👍

**Pull request merged**